sklearn/datasets/_kddcup99.py

```diff
@@ -386,12 +386,14 @@ def _fetch_brute_kddcup99(
         DT = np.dtype(dt)
         logger.debug("extracting archive")
         archive_path = join(kddcup_dir, archive.filename)
-        file_ = GzipFile(filename=archive_path, mode="r")
         Xy = []
-        for line in file_.readlines():
-            line = line.decode()
-            Xy.append(line.replace("\n", "").split(","))
-        file_.close()
+
+        # Opening file using context manager to ensure file is always closed
```

Comment:
I'd remove this, use of a context manager for file operations is common place now, so there is no need to explain what the line of code below does (same for the `Image.open` case)
```suggestion
```